source/kinsol_interface.cc

```diff
@@ -0,0 +1,257 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2015 by the deal2lkit authors
+//
+//    This file is part of the deal2lkit library.
+//
+//    The deal2lkit library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal2lkit distribution.
+//
+//-----------------------------------------------------------
+
+#include <deal2lkit/kinsol_interface.h>
+#ifdef D2K_WITH_SUNDIALS
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/generic_linear_algebra.h>
+
+#ifdef DEAL_II_WITH_PETSC
+#include <deal.II/lac/petsc_solver.h>
+#ifdef DEAL_II_WITH_MPI
+#include <deal.II/lac/petsc_parallel_vector.h>
+#endif
+#endif
+
+#ifdef DEAL_II_WITH_TRILINOS
+#include <deal.II/lac/trilinos_block_vector.h>
+#ifdef DEAL_II_WITH_MPI
+#include <deal.II/lac/trilinos_parallel_block_vector.h>
+#endif
+#endif
+
+#include <deal2lkit/utilities.h>
+
+D2K_NAMESPACE_OPEN
+
+
```

Comment:
wrap this function into an anonymous namespace. In this way, only this file will ever see this function.